### DISCUSSION TITLE
Please explain GenerateTexture() in more depth

### DISCUSSION THREAD

**Discussion by MrMilad** [mikke89/RmlUi#287] [Q&A] [answered]
Hi, please explain when and exactly why this function is called and what are the purposes of the created textures? does its calling frequency depend on something? if yes, what?
I checked source code but couldn't find anything good due to lack of time and large code base.
This function will affect our render pipeline and resource management system due to texture creating/switching. if we silently ignore this function, what is the side effect?

**Comment by mikke89** [comment, marked as answer]
Hi.

This function is generally called for generated font atlases. So, you wouldn't see any text if you ignored it. If it really is a problem for the pipeline, you may consider writing a custom font engine based on pre-rendered atlases. There is a sample for this so you can take a look there.
